### PR TITLE
Fixed comments on PR #9735

### LINKED ISSUE #10159: "Truncate hostnames to 15 characters for NetBIOS compatibility." checkbox is not clickable when creating an instance.
**Setup**
- Rancher version: 2.8.1
- Rancher UI Extensions: N/A
- Browser type & version: Chrome

**Describe the bug**

"Truncate hostnames to 15 characters for NetBIOS compatibility" checkbox cannot be checked.

**To Reproduce**

Clusters -> Create -> Network

**Result**
"Truncate hostnames to 15 characters for NetBIOS compatibility" cannot be checked.

**Expected Result**

"Truncate hostnames to 15 characters for NetBIOS compatibility" should be clickable.

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by eva-vashkevich**
### Summary
Fixes #10159 - has to compare with master because this bug has been introduced after 2.8.0 release.
Fixed comments from https://github.com/rancher/dashboard/pull/9735

### Occurred changes and/or fixed issues

- Fixed "Truncate hostnames to 15 characters for NetBIOS compatibility." checkbox
- Removed truncateHostnames property from rke2 file and renamed truncateName function to truncateHostname
- Fixed the truncateLimit logic
- Separated new tabs into a separate folder. (I think this will need to be changed but I don't have a better structure atm)
- Moved yaml refresh back to when the tab becomes active
- removed unneeded isK3s, get, isView, and clusterIsAlreadyCreated props

### Areas or cases that should be tested

Networking, Basics, AddOnConfig tabs should be tested

### Areas which could experience regressions

Cluster creation and edit, cluster import

### CODE REVIEW

File: shell/edit/provisioning.cattle.io.cluster/tabs/Etcd.vue [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { LabeledInput } from '@components/Form/LabeledInput';
 
 import { RadioGroup } from '@components/Form/Radio';
 
-import S3Config from './S3Config';
+import S3Config from '@shell/edit/provisioning.cattle.io.cluster/S3Config';
```
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/tabs/Etcd.vue:6]: this is only used from this component. to keep things neat in the prov cluster folder can the same be done for this?

provisioning.cattle.io.cluster/tabs/etcd/index.vue
provisioning.cattle.io.cluster/tabs/etcd/S3Config.vue

File: shell/edit/provisioning.cattle.io.cluster/tabs/Networking.vue [hunk 2]
```diff
@@ -2,9 +2,9 @@
 import { LabeledInput } from '@components/Form/LabeledInput';
 import { Banner } from '@components/Banner';
 import { Checkbox } from '@components/Form/Checkbox';
-import { _EDIT } from '@shell/config/query-params';
+import { _EDIT, _VIEW } from '@shell/config/query-params';
 import ArrayList from '@shell/components/form/ArrayList';
-import ACE from './ACE';
+import ACE from '@shell/edit/provisioning.cattle.io.cluster/ACE';
```
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/tabs/Networking.vue:7]: same as per etcd and S3Config,  this ACE can move in to it's parent components folder

**Review by richard-cox — CHANGES_REQUESTED** [review]
- one re-opened comment https://github.com/rancher/dashboard/pull/9735#discussion_r1431632508
- 4 similar requests to move components only used by their tab components

**Review by aalves08 — APPROVED** [review]
Testing and codewise, LGTM @eva-vashkevich , so I think this can go in.

For another issue, perhaps, but I've seen that the `additional manifest` doesn't reset if you change the `Container Network` under `Basics` tab. I don't know if this is the correct behaviour or not.

https://github.com/rancher/dashboard/assets/97888974/a58eb330-81c8-4be0-a6ef-75d91bc1a671

This was already happening on `v2.8.0` so I don't think it's a problem for this PR. 

What's your opinion @nwmac or @richard-cox ?

**Pull request merged**